ReleaseNotes/ThunderReleaseNotes_R5.2.md

```diff
@@ -0,0 +1,186 @@
+
+### Change: Thunder Meta data
+
+The Thunder meta data (that can be retrieved via a call to the Controller) was extended and now contains more info (mainly intended for QA and debug purposes)
+
+### Change: Process container updates
+
+Some flavours of the ProcessContainer implementations were updated to reflect the changes made in Thunder 5.1 where the ability to run multiple container flavour implementation at runtime was added.
+
+### Change: JSONRPCLink improvements
+
+The JSONRPCLink now also makes sure events it was subscribed on for will be restored when a connection was lost and restored. 
+
+### Change: Windows socket handling improvements
+
+The Windows socket handling code was improved (e.g. in case an overlapped IO operation was reported)
+
+### Feature: Custom JSONRPC error messages and error codes supported.
+
+Although not advised as it will break consistency and generated documentation it is now possible to override the default errormessage and errorcode for JSON-RPC when generated from the COM-RPC method call return code. See [here](https://github.com/rdkcentral/ThunderNanoServices/tree/master/examples/DynamicJSONRPCErrorMessage) for an example how it can be used.
+
+
+### Change: General bug fixes
+
+A number of issues found in Thunder 5.1 have been corrected in Thunder 5.2. 
+
+## Breaking Changes
+
+Although Thunder 5.2 is a minor upgrade and therefore should not contain breaking changes there is one however. This because of a request to the Thunder team to change the behaviour of the ThunderTooling in this regard. 
+In itself this change in the ThunderTooling should not lead to different behaviour in Interfaces used in RDKServices as far as we are aware. The old Thunder Interfaces used in RDKServices were all adapted to generate the exact same interface as in 5.1 and others used different @text options already to force the behaviour that has now become the default.
+
+### default case for generated JSON-RPC code.
+
+The ThunderTools now generate a different casing for the JSON-RPC code by default then it did before. Thunder 5.1 and older generated lowercase function, parameters, PoD members and event names by default from the interface files while the enums where PascalCase by default. Thunder 5.2 generates camelCase function, parameters, PoD members and event names by default while the enums are UPPER_SNAKE by default.
+See for more details below and also on new options to influence the generated case.
+
+# Thunder Tools
+
+## Changes and new Features 
+
+### Feature: JSON-RPC case options
+
+Up to now the JSON-RPC code generators generated lowercase function, parameters, PoD members and event names by default from the interface files while the enums where PascalCase by default. Quite a number of interfaces used different @text options to deviate from this and it was decided to make camelCase the default for function function, parameters, PoD members and event names while using UPPER_SNAKE for enums. 
```

Comment:
Nit: "camelCase the default for function function, parameters," - comma in wrong place?

Reply:
Fixed